TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -275,8 +277,17 @@ public async Task<CommandResult> Pokedex(CommandContext context)
             {
                 RegionInformation regionInformation = _pokedexModeRegions[mode];
                 ImmutableSortedDictionary<PkmnSpecies, int> ownedPokemons = (await _badgeRepo.CountByUserPerSpecies(user.Id));
-                int userOwnedRegionCount = ownedPokemons.Count(ownedPokemon =>
-                    ownedPokemon.Key.GetGeneration() == regionInformation.Generation);
+                int userOwnedRegionCount;
+                if (mode.Equals(PokedexModeNational))
+                {
+                    userOwnedRegionCount = ownedPokemons.Count(ownedPokemon =>
+                        ownedPokemon.Key.GetGeneration() != regionInformation.Generation);
+                }
+                else
+                {
+                    userOwnedRegionCount = ownedPokemons.Count(ownedPokemon =>
+                        ownedPokemon.Key.GetGeneration() == regionInformation.Generation);
+                }
```

Comment:
sure, but can you make that more compact?
```suggestion
                int userOwnedRegionCount = mode.Equals(PokedexModeNational)
                    ? ownedPokemons.Count(ownedPokemon => ownedPokemon.Key.GetGeneration() != regionInformation.Generation)
                    : ownedPokemons.Count(ownedPokemon => ownedPokemon.Key.GetGeneration() == regionInformation.Generation);
```